src/hooks/useInput.js

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useRef, useState } from 'react'
+import { useFocusEnter, useFocusLeave } from '@aragon/ui'
+
+export function useInput(validationFn) {
+  const [value, setValue] = useState('')
+  const [status, setStatus] = useState('empty')
+
+  const inputRef = useRef()
+
+  const handleBlur = useCallback(() => {
+    const invalidValue = !validationFn(value)
+    if (invalidValue) {
+      return setStatus('invalid')
+    }
+    return setStatus('valid')
+  }, [validationFn, value])
+
+  const handleInputFocus = useCallback(event => event.target.select(), [])
```

Comment:
I'm not sure if this is consistent with how we use other inputs; might want to double check with @dizzypaty if we should select the text when focusing.

Reply:
I wonder if it should be here, even if we want this behavior? What would you guys think of renaming it `useInputValidation()` and keeping it specialized?

Reply:
should i remove the input focus function maybe?

Reply:
I think we’ll still need it for https://github.com/aragon/court-dashboard/pull/343/files#r430469262